pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/GenericRecordWrapper.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.sink.pulsar;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pulsar.client.api.schema.Field;
+import org.apache.pulsar.client.api.schema.GenericRecord;
+
+/** Generic record for wrapping nested native objects like Maps and Lists */
+public class GenericRecordWrapper implements GenericRecord {
+  private final Map<String, Object> values;
+  private final List<Field> fields;
+  private final Object nativeObject;
+
+  public GenericRecordWrapper(Object nativeObject) {
+    if (nativeObject == null) throw new NullPointerException("nativeObject cannot be null!");
+
+    this.values = new LinkedHashMap<>();
+    this.fields = new ArrayList<>();
+    this.nativeObject = nativeObject;
+  }
+
+  @Override
+  public byte[] getSchemaVersion() {
+    return new byte[1];
```

Comment:
this should be byte[0]

Reply:
and we can use a constant (I am not sure if this method will ever be called, if it is never called then we can throw UnsupportedOperationException

Reply:
+1. I only wanted this to wrap a native object, an exception would be reasonable.